cmd/system-probe/utils/memory_monitor_linux.go

```diff
@@ -69,13 +69,16 @@ func getActionCallback(action string) (func(), string, error) {
 				log.Errorf("Failed to generate memory profile: %s", err)
 				return
 			}
-			defer memProfile.Close()
 
 			if err := pprof.WriteHeapProfile(memProfile); err != nil {
 				log.Errorf("Failed to generate memory profile: %s", err)
 				return
```

Comment:
The file will fail to be closed if this error condition happens.

Reply:
Nice catch ! Thank you. I fixed it in my last commit.